src/main/java/fi/otavanopisto/kuntaapi/server/integrations/management/ManagementNewsArticleEntityUpdater.java

```diff
@@ -102,62 +93,34 @@ public void startTimer() {
   }
 
   private void startTimer(int duration) {
-    stopped = false;
     TimerConfig timerConfig = new TimerConfig();
     timerConfig.setPersistent(false);
     timerService.createSingleActionTimer(duration, timerConfig);
   }
-
-  @Override
-  public void stopTimer() {
-    stopped = true;
-  }
   
-  @Asynchronous
-  public void onOrganizationIdUpdateRequest(@Observes NewsArticleIdUpdateRequest event) {
-    if (!stopped) {
-      NewsArticleId newsArticleId = event.getId();
-      
-      if (!StringUtils.equals(newsArticleId.getSource(), ManagementConsts.IDENTIFIER_NAME)) {
-        return;
-      }
-      
-      queue.add(event);
-    }
-  }
-  
-  @Asynchronous
-  public void onNewsArticleIdRemoveRequest(@Observes NewsArticleIdRemoveRequest event) {
-    if (!stopped) {
-      NewsArticleId newsArticleId = event.getId();
-      
-      if (!StringUtils.equals(newsArticleId.getSource(), ManagementConsts.IDENTIFIER_NAME)) {
-        return;
-      }
-      
-      deleteNewsArticle(event, newsArticleId);
+  @Timeout
+  public void timeout(Timer timer) {
+    if (systemSettingController.isNotTestingOrTestRunning()) {
+      executeNextTask();
     }
+
+    startTimer(systemSettingController.inTestMode() ? 1000 : TIMER_INTERVAL);
   }
   
-  @Timeout
-  public void timeout(Timer timer) {
-    if (!stopped) {
-      if (systemSettingController.isNotTestingOrTestRunning()) {
-        NewsArticleIdUpdateRequest updateRequest = queue.next();
-        if (updateRequest != null) {
-          updateManagementPost(updateRequest);
-        }
+  private void executeNextTask() {
+    IdTask<NewsArticleId> task = newsArticleIdTaskQueue.next();
+    if (task != null) {
+      if (task.getOperation() == Operation.UPDATE) {
+        updateManagementPost(task.getId(), task.getOrderIndex()); 
+      } else if (task.getOperation() == Operation.REMOVE) {
+        deleteManagementPost(task.getId());
       }
-
-      startTimer(systemSettingController.inTestMode() ? 1000 : TIMER_INTERVAL);
     }
   }
   
-  private void updateManagementPost(NewsArticleIdUpdateRequest updateRequest) {
-    OrganizationId organizationId = updateRequest.getOrganizationId();
-    NewsArticleId newsArticleId = updateRequest.getId();
+  private void updateManagementPost(NewsArticleId newsArticleId, Long orderIndex) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused private "updateManagementPost" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)